src/uid2Sdk.ts

```diff
@@ -39,38 +46,29 @@ export abstract class UID2SdkBase {
   // Dependencies initialised on construction
   private _logger: Logger;
   private _tokenPromiseHandler: UID2PromiseHandler;
-  private _callbackManager: Uid2CallbackManager;
+  protected _callbackManager: Uid2CallbackManager;
 
   // Dependencies initialised on call to init due to requirement for options
   private _storageManager: UID2StorageManager | undefined;
   private _apiClient: Uid2ApiClient | undefined;
 
   // State
-  private _product: Product;
+  private _product: ProductDetails;
   private _opts: Uid2Options = {};
   private _identity: Uid2Identity | null | undefined;
   private _initComplete = false;
 
-  constructor(
+  // Sets up nearly everything, but does not run SdkLoaded callbacks - derived classes must run them.
+  protected constructor(
     existingCallbacks: Uid2CallbackHandler[] | undefined = undefined,
-    callbackContainer: CallbackContainer,
-    product: Product
+    product: ProductDetails
   ) {
     this._product = product;
-    this._logger = MakeLogger(console, product);
+    this._logger = MakeLogger(console, product.name);
     if (existingCallbacks) this.callbacks = existingCallbacks;
 
     this._tokenPromiseHandler = new UID2PromiseHandler(this);
     this._callbackManager = new Uid2CallbackManager(this, () => this.getIdentity(), this._logger);
-    const runCallbacks = () => {
-      this._callbackManager.runCallbacks(EventType.SdkLoaded, {});
-    };
-    if (window.__uid2 instanceof UID2) {
```

Comment:
Can't do this in the base class because of this - each derived class will need to do something like this. Maybe I could find a way to do this without adding a responsibility to the deriving class but it doesn't seem worth it.